File: Oriel_Calculated_Sharp/MultiverseLinkedList.cs
using System;

public class Node
{
    public string Spiderman { get; set; }
    public Node Next { get; set; }

    public Node(string spiderman)
    {
        Spiderman = spiderman;
        Next = null;
    }
}

public class MultiverseNodeCreator
{
    private Node head;
    private Node tail;
    private int length;

    public MultiverseNodeCreator()
    {
        head = null;
        tail = null;
        length = 0;
    }

    public void Push(string spiderman)
    {
        Node newSpiderman = new Node(spiderman);
        if (head == null || tail == null)
        {
            head = newSpiderman;
            tail = head;
        }
        else
        {
            tail.Next = newSpiderman;
            tail = newSpiderman;
        }
        length++;
    }

    public string Pop()
    {
        if (head == null)
        {
            return null;
        }

        Node currentSpiderman = head;
        Node previousSpiderman = null;

        while (currentSpiderman.Next != null)
        {
            previousSpiderman = currentSpiderman;
            currentSpiderman = currentSpiderman.Next;
        }

        tail = previousSpiderman;

        if (previousSpiderman == null)
        {
            head = null;
        }
        else
        {
            previousSpiderman.Next = null;
        }

        length--;
        return currentSpiderman.Spiderman;
    }

    public string Shift()
    {
        if (head == null)
        {
            return null;
        }

        Node shiftedSpiderman = head;
        head = head.Next;
        length--;

        if (head == null)
        {
            tail = null;
        }

        return shiftedSpiderman.Spiderman;
    }

    public void Unshift(string spiderman)
    {
        Node newSpiderman = new Node(spiderman);
        if (head == null)
        {
            head = newSpiderman;
            tail = head;
        }
        else
        {
            newSpiderman.Next = head;
            head = newSpiderman;
        }
        length++;
    }
}

File: Oriel_whasJava_Script/MultiverseLinkedList.js


class Node {
    constructor(spiderman) {
        this.spiderman = spiderman;
        this.next = null;
    };
};


class MultiverseNodeCreator {
    constructor() {
    this.head = null;
    this.tail = null;
    this.length = 0;
    };

    push(spiderman) {
        let newSpiderMan = new Node(spiderman)
        if (!this.head || !this.tail) {
            this.head = newSpiderMan;
            this.tail = this.head;
        } else {
            this.tail.next = newSpiderMan;
            this.tail = newSpiderMan
        };
        this.length++;
    };

    pop() {
        if (!this.head) {
            return null;
        };

        let currentSpiderman = this.head;
        let previousSpiderman = null;

        while (currentSpiderman.next) {
            previousSpiderman = currentSpiderman;
            currentSpiderman = currentSpiderman.next;
        };

        this.tail = previousSpiderman;

        if (!previousSpiderman) {
            this.head = null;
        } else {
            previousSpiderman.next = null;
        };

        this.length--;
        return currentNode.spiderman;
    };


    shift() {
        if (!this.head) {
            return null;
        };

        let shiftedSpiderman = this.head;
        this.head = this.head.next;
        this.length--;

        if (!this.head) {
            this.tail = null;
        };

        return shiftedSpiderman.spiderman;
    };


    unshift(spiderman) {
        let newSpiderMan = new Node(spiderman);
        if (!this.head) {
            this.head = newSpiderMan;
            this.tail = this.head;
        } else {
            newSpiderMan.next = this.head;
            this.head = newSpiderMan;
        }
        this.length++;
    };
};
